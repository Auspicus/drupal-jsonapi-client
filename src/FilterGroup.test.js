/* eslint-disable import/first */
require('regenerator-runtime/runtime') // eslint-disable-line

import { Request, Response } from 'node-fetch'
import FilterGroup from './FilterGroup'
import Filter from './Filter'

global.Request = Request
global.Response = Response

describe('Filter', () => {
  it('serializes filter group with conjunction AND to a query', () => {
    expect(new FilterGroup({
      identifier: 'group-1',
      type: 'AND',
      children: [
        new Filter({
          identifier: 'filter-1',
          path: 'title',
          value: 'asdf123',
        }),
        new Filter({
          identifier: 'filter-2',
          path: 'status',
          value: 1,
        }),
      ],
    }).query()).toMatchSnapshot()
  })

  it('serializes filter group with conjunction OR to a query', () => {
    expect(new FilterGroup({
      identifier: 'group-1',
      type: 'OR',
      children: [
        new Filter({
          identifier: 'filter-1',
          path: 'title',
          value: 'asdf123',
        }),
        new Filter({
          identifier: 'filter-2',
          path: 'status',
          value: 1,
        }),
      ],
    }).query()).toMatchSnapshot()
  })

  it('serializes nested filter groups', () => {
    expect(new FilterGroup({
      identifier: 'group-3',
      type: 'OR',
      children: [
        new FilterGroup({
          identifier: 'group-1',
          type: 'AND',
          children: [
            new Filter({
              identifier: 'filter-1',
              path: 'title',
              value: 'asdf123',
            }),
            new Filter({
              identifier: 'filter-2',
              path: 'status',
              value: 1,
            }),
          ],
        }),
        new FilterGroup({
          identifier: 'group-2',
          type: 'AND',
          children: [
            new Filter({
              identifier: 'filter-3',
              path: 'title',
              value: 'asdf123',
            }),
            new Filter({
              identifier: 'filter-4',
              path: 'status',
              value: 1,
            }),
          ],
        }),
      ],
    }).query()).toMatchSnapshot()
  })
})
