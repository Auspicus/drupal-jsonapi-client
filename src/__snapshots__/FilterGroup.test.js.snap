// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter serializes filter group with conjunction AND to a query 1`] = `
Array [
  "filter[group-1][group][conjunction]=AND",
  "",
  Array [
    "filter[filter-1][condition][path]=title",
    "",
    "filter[filter-1][condition][memberOf]=group-1",
    "filter[filter-1][condition][value]=asdf123",
  ],
  Array [
    "filter[filter-2][condition][path]=status",
    "",
    "filter[filter-2][condition][memberOf]=group-1",
    "filter[filter-2][condition][value]=1",
  ],
]
`;

exports[`Filter serializes filter group with conjunction OR to a query 1`] = `
Array [
  "filter[group-1][group][conjunction]=OR",
  "",
  Array [
    "filter[filter-1][condition][path]=title",
    "",
    "filter[filter-1][condition][memberOf]=group-1",
    "filter[filter-1][condition][value]=asdf123",
  ],
  Array [
    "filter[filter-2][condition][path]=status",
    "",
    "filter[filter-2][condition][memberOf]=group-1",
    "filter[filter-2][condition][value]=1",
  ],
]
`;

exports[`Filter serializes nested filter groups 1`] = `
Array [
  "filter[group-3][group][conjunction]=OR",
  "",
  Array [
    "filter[group-1][group][conjunction]=AND",
    "filter[group-1][group][memberOf]=group-3",
    Array [
      "filter[filter-1][condition][path]=title",
      "",
      "filter[filter-1][condition][memberOf]=group-1",
      "filter[filter-1][condition][value]=asdf123",
    ],
    Array [
      "filter[filter-2][condition][path]=status",
      "",
      "filter[filter-2][condition][memberOf]=group-1",
      "filter[filter-2][condition][value]=1",
    ],
  ],
  Array [
    "filter[group-2][group][conjunction]=AND",
    "filter[group-2][group][memberOf]=group-3",
    Array [
      "filter[filter-3][condition][path]=title",
      "",
      "filter[filter-3][condition][memberOf]=group-2",
      "filter[filter-3][condition][value]=asdf123",
    ],
    Array [
      "filter[filter-4][condition][path]=status",
      "",
      "filter[filter-4][condition][memberOf]=group-2",
      "filter[filter-4][condition][value]=1",
    ],
  ],
]
`;
